test/interpreter/function-binom.inv.spec.ts

```diff
@@ -0,0 +1,165 @@
+        '=BINOM.INV(0, 0.8, 0.6)',
+        '=BINOM.INV(0, 0.8, 0.7)',
+        '=BINOM.INV(0, 0.8, 0.8)',
+        '=BINOM.INV(0, 0.8, 0.9)',
+        '=BINOM.INV(0, 0.8, 0.999)'],
+    ])
+
+    expect(engine.getSheetValues(0)).toEqual([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]])
+  })
+
+  it('should work, another small number of trials', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=BINOM.INV(1, 0.8, 0.001)',
+        '=BINOM.INV(1, 0.8, 0.1)',
+        '=BINOM.INV(1, 0.8, 0.2)',
+        '=BINOM.INV(1, 0.8, 0.3)',
+        '=BINOM.INV(1, 0.8, 0.4)',
+        '=BINOM.INV(1, 0.8, 0.5)',
+        '=BINOM.INV(1, 0.8, 0.6)',
+        '=BINOM.INV(1, 0.8, 0.7)',
+        '=BINOM.INV(1, 0.8, 0.8)',
+        '=BINOM.INV(1, 0.8, 0.9)',
+        '=BINOM.INV(1, 0.8, 0.999)'],
+    ])
+
+    expect(engine.getSheetValues(0)).toEqual([[0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1]])
+    //both products #1 and #2 return 0 for '=BINOM.INV(1, 0.8, 0.2)', which is incorrect
+  })
+
+  it('should work, large number of trials', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=BINOM.INV(1000, 0.8, 0.001)',
+        '=BINOM.INV(1000, 0.8, 0.1)',
+        '=BINOM.INV(1000, 0.8, 0.2)',
+        '=BINOM.INV(1000, 0.8, 0.3)',
+        '=BINOM.INV(1000, 0.8, 0.4)',
+        '=BINOM.INV(1000, 0.8, 0.5)',
+        '=BINOM.INV(1000, 0.8, 0.6)',
+        '=BINOM.INV(1000, 0.8, 0.7)',
+        '=BINOM.INV(1000, 0.8, 0.8)',
+        '=BINOM.INV(1000, 0.8, 0.9)',
+        '=BINOM.INV(1000, 0.8, 0.999)'],
+    ])
+
+    expect(engine.getSheetValues(0)).toEqual([[760, 784, 789, 793, 797, 800, 803, 807, 811, 816, 838]])
+  })
+
+  it('truncation works', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=BINOM.INV(1000.1, 0.8, 0.001)',
+        '=BINOM.INV(1000.2, 0.8, 0.1)',
+        '=BINOM.INV(1000.3, 0.8, 0.2)',
+        '=BINOM.INV(1000.4, 0.8, 0.3)',
+        '=BINOM.INV(1000.5, 0.8, 0.4)',
+        '=BINOM.INV(1000.6, 0.8, 0.5)',
+        '=BINOM.INV(1000.7, 0.8, 0.6)',
+        '=BINOM.INV(1000.8, 0.8, 0.7)',
+        '=BINOM.INV(1000.9, 0.8, 0.8)',
+        '=BINOM.INV(1000.99, 0.8, 0.9)',
+        '=BINOM.INV(1000.999, 0.8, 0.999)'],
+    ])
+
+    expect(engine.getSheetValues(0)).toEqual([[760, 784, 789, 793, 797, 800, 803, 807, 811, 816, 838]])
+  })
+
+  it('checks bounds', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=BINOM.INV(0, 0.5, 0.5)'],
+      ['=BINOM.INV(-0.001, 0.5, 0.5)'],
+      ['=BINOM.INV(10, 0, 0.5)'],
+      ['=BINOM.INV(10, 1, 0.5)'],
+      ['=BINOM.INV(10, -0.001, 0.5)'],
+      ['=BINOM.INV(10, 1.001, 0.5)'],
+      ['=BINOM.INV(10, 0.5, 0)'],
+      ['=BINOM.INV(10, 0.5, 1)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(0)
+    expect(engine.getCellValue(adr('A2'))).toEqualError(detailedError(ErrorType.NUM, ErrorMessage.ValueSmall))
+    expect(engine.getCellValue(adr('A3'))).toEqual(0)
+    expect(engine.getCellValue(adr('A4'))).toEqual(10)
```

Comment:
I got #NUM! error for both in XL and GS so they are correctly classified as bounds, but calculate a value for them?

Reply:
Judging by the docs, it should be computed:
https://support.microsoft.com/en-us/office/binom-inv-function-80a0370c-ada6-49b4-83e7-05a91ba77ac9

and by the definition, it also should be computed. My guess is that the implementation in XL/GS has rounding errors in BINOM.DIST so inverting this function at the border of domain results in an error. I'll add appropriate comment.

Reply:
OK 👍